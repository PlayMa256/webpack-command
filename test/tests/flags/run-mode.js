const { apply, build, crcDist, test, validate } = require('../../util');

test('--run-dev', module, () => {
  const fixture = 'run-mode/run-dev';
  const opts = { fixture };

  let config;

  it(`should validate`, () => {
    expect(validate(opts)).toEqual(true);
  });

  it(`should apply`, () => {
    config = apply(opts);

    expect(config).toMatchSnapshot();
  });

  it(`should build`, () =>
    build(config).then((result) => {
      expect(result).toMatchSnapshot();
      expect(crcDist()).toMatchSnapshot();
    }));
});

test('--run-prod', module, () => {
  const fixture = 'run-mode/run-prod';
  const opts = { fixture };

  let config;

  it(`should validate`, () => {
    expect(validate(opts)).toEqual(true);
  });

  it(`should apply`, () => {
    config = apply(opts);

    expect(config).toMatchSnapshot();
  });

  it(`should build`, () =>
    build(config).then((result) => {
      expect(result).toMatchSnapshot();
      expect(crcDist()).toMatchSnapshot();
    }));
});
