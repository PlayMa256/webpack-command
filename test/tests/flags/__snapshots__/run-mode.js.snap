// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Flags > --run-dev > should apply #0 1`] = `
Object {
  "context": "<PROJECT_ROOT>",
  "devtool": "source-map",
  "entry": "<PROJECT_ROOT>/test/fixtures/common/entry-a.js",
  "mode": "development",
  "plugins": Array [
    NamedModulesPlugin {
      "options": Object {},
    },
  ],
  "reporter": "<PROJECT_ROOT>/test/fixtures/common/test-reporter.js",
  "stats": Object {
    "builtAt": false,
    "hash": false,
    "timings": false,
  },
}
`;

exports[`Flags > --run-dev > should build #0 1`] = `
"Version: webpack 4.3.0
      Asset      Size  Chunks             Chunk Names
    main.js  2.92 KiB    main  [emitted]  main
main.js.map  2.66 KiB    main  [emitted]  main
Entrypoint main = main.js main.js.map
[./test/fixtures/common/entry-a.js] 41 bytes {main} [built]"
`;

exports[`Flags > --run-dev > should build #1 1`] = `"f349b18e"`;

exports[`Flags > --run-prod > should apply #0 1`] = `
Object {
  "context": "<PROJECT_ROOT>",
  "devtool": "source-map",
  "entry": "<PROJECT_ROOT>/test/fixtures/common/entry-a.js",
  "mode": "development",
  "plugins": Array [
    NamedModulesPlugin {
      "options": Object {},
    },
  ],
  "reporter": "<PROJECT_ROOT>/test/fixtures/common/test-reporter.js",
  "stats": Object {
    "builtAt": false,
    "hash": false,
    "timings": false,
  },
}
`;

exports[`Flags > --run-prod > should build #0 1`] = `
"Version: webpack 4.3.0
      Asset      Size  Chunks             Chunk Names
    main.js  2.92 KiB    main  [emitted]  main
main.js.map  2.66 KiB    main  [emitted]  main
Entrypoint main = main.js main.js.map
[./test/fixtures/common/entry-a.js] 41 bytes {main} [built]"
`;

exports[`Flags > --run-prod > should build #1 1`] = `"f349b18e"`;
